### PR TITLE
improve mx UI; move valid-mx parameters to UI too

### PR DESCRIPTION
**Pull request by gawel**

### CODE REVIEW

File: modoboa/admin/templates/admin/mx_domain_detail.html [hunk 1]
```diff
@@ -15,14 +15,27 @@
       {% for mx in domain.mxrecord_set.all %}
         <tr>
           <td>{{ mx.name }}</td>
-          <td>{{ mx.address }}</td>
+          <td>
+            {% if mx.is_managed %}
+              {% if not mx.managed %}
+                <a class="label label-danger"
+                   href="{% url 'core:index' %}#parameters"
```
Comment (gawel) [inline modoboa/admin/templates/admin/mx_domain_detail.html]: This sucks. Do you have a clever way to point to the right section ?

Comment (tonioo) [inline modoboa/admin/templates/admin/mx_domain_detail.html]: No...

Comment (gawel) [inline modoboa/admin/templates/admin/mx_domain_detail.html]: Link removed. Good to merge (for me)

File: modoboa/admin/app_settings.py [hunk 2]
```diff
@@ -61,6 +61,17 @@ class AdminParametersForm(parameters.AdminParametersForm):
         )
     )
 
+    valid_mxs = forms.CharField(
+        label=ugettext_lazy("Valid MXs"),
+        initial="",
+        help_text=ugettext_lazy(
+            "IP(s) or netmask(s) "
```
Comment (tonioo) [inline modoboa/admin/app_settings.py]: netmask or network ?

File: modoboa/admin/management/commands/subcommands/_mx.py [hunk 3]
```diff
@@ -38,13 +38,16 @@ def sender(self):
         """Return sender address for notifications."""
         return parameters.get_admin("SENDER_ADDRESS", app="core")
 
+    @cached_property
+    def valid_mxs(self):
+        """Return valid MXs set in admin."""
+        valid_mxs = parameters.get_admin("VALID_MXS", app="admin")
+        return [ipaddress.ip_network(u"{}".format(v.strip()))
+                for v in valid_mxs.split('\n') if v.strip()]
```
Comment (tonioo) [inline modoboa/admin/management/commands/subcommands/_mx.py]: Looks like you are/were a C programmer :D

File: modoboa/admin/models/domain.py [hunk 4]
```diff
@@ -224,6 +224,17 @@ def post_create(self, creator):
 reversion.register(Domain)
 
 
+class MXQuerySet(models.QuerySet):
+    """Custom manager for MXRecord."""
+
+    def valids(self):
+        """Return managed results."""
+        if parameters.get_admin("ENABLE_MX_CHECKS"):
+            if parameters.get_admin("VALID_MXS", app="admin").strip():
```
Comment (tonioo) [inline modoboa/admin/models/domain.py]: Is it really necessary to check those settings in the manager ?

File: modoboa/admin/models/domain.py [hunk 5]
```diff
@@ -224,6 +224,17 @@ def post_create(self, creator):
...
+                return self.filter(managed=True).exists()
+        return self.exists()
```
Comment (tonioo) [inline modoboa/admin/models/domain.py]: So you just check is there is manager MXs in the database ? You don't want to retrieve them with this method ?

File: modoboa/admin/app_settings.py (hunk 2 above)
Comment (gawel) [inline modoboa/admin/app_settings.py]: network

File: modoboa/admin/management/commands/subcommands/_mx.py (hunk 3 above)
Comment (gawel) [inline modoboa/admin/management/commands/subcommands/_mx.py]: Not at all

File: modoboa/admin/models/domain.py (hunk 4 above)
Comment (gawel) [inline modoboa/admin/models/domain.py]: yes, unless you always want to show some errors

File: modoboa/admin/models/domain.py (hunk 5 above)
Comment (gawel) [inline modoboa/admin/models/domain.py]: yep

File: modoboa/admin/management/commands/subcommands/_mx.py (hunk 3 above)
Comment (tonioo) [inline modoboa/admin/management/commands/subcommands/_mx.py]: So why do you put single quotes around string with only one character ? ;)

File: modoboa/admin/models/domain.py (hunk 4 above)
Comment (tonioo) [inline modoboa/admin/models/domain.py]: I guess we could disable this access earlier ?

Comment (gawel) [inline modoboa/admin/models/domain.py]: well no, you want to show the buttons in both cases.

**Pull request merged**